### PR TITLE
fix: adapt jupyter notebook codes to 2.0

### LINKED ISSUE #2899: Minimum Working Example Jina 2.0 - Name Error, __file__ not found
**Describe the bug**
Hi, the minimum working example for Jina 2.0 does not fully run in Colab. Specifically, the flow cell throws a name error as it cannot find the __file__, which is the source code itself.

I'm referring to:
https://colab.research.google.com/github/jina-ai/jupyter-notebooks/blob/main/2.0-readme-38l.ipynb

### PR DESCRIPTION
**Pull request by Kelton8Z**
Closes [#2899](https://github.com/jina-ai/jina/issues/2899)

### CODE REVIEW

File: 2.0-readme-38l.ipynb [hunk 1]
```diff
@@ -187,85 +250,106 @@
...
+      "\u001b[0;31mIsADirectoryError\u001b[0m: [Errno 21] Is a directory: '/Users/kel/codes/jina/jupyter-notebooks'"
+     ]
+    }
+   ],
+   "source": [
+    "f = Flow(port_expose=12345).add(uses=CharEmbed, parallel=2).add(uses=Indexer)  # build a Flow, with 2 parallel CharEmbed, tho unnecessary\n",
+    "with f:\n",
+    "    f.post('/index', (Document(text=t.strip()) for t in open(os.getcwd()) if t.strip()))  # index all lines of this file\n",
```
Comment (cristianmtr) [inline 2.0-readme-38l.ipynb]: how would you read lines from a directory?

```
IsADirectoryError                         Traceback (most recent call last)
<ipython-input-6-d29dc520f082> in <module>
      1 f = Flow(port_expose=12345).add(uses=CharEmbed, parallel=2).add(uses=Indexer)  # build a Flow, with 2 parallel CharEmbed, tho unnecessary
      2 with f:
----> 3     f.post('/index', (Document(text=t.strip()) for t in open(os.getcwd()) if t.strip()))  # index all lines of this file
      4     f.block()  # block for listening request

IsADirectoryError: [Errno 21] Is a directory: '/home/cristian/code/jupyter-notebooks
```

**Review by cristianmtr — CHANGES_REQUESTED** [review]

**Comment by cristianmtr** [comment]
also fix the issue with f.block

File: 2.0-readme-38l.ipynb (hunk 1 above)
Comment (Kelton8Z) [reply 2.0-readme-38l.ipynb]: changed to `os.__file__`

**Comment by Kelton8Z** [comment]
> also fix the issue with f.block

seems that the current code can't make the search request

File: basic-basic-crud-functions.ipynb [hunk 2]
```diff
@@ -38,8 +38,8 @@
    "outputs": [],
    "source": [
     "import numpy as np\n",
-    "from jina import Document\n",
-    "docs = [Document(id='🐲', embedding=np.array([0, 0]), tags={'guardian': 'Azure Dragon', 'position': 'East'}),\n",
+    "from jina import Document, DocumentArray\n",
+    "docs = DocumentArray[Document(id='🐲', embedding=np.array([0, 0]), tags={'guardian': 'Azure Dragon', 'position': 'East'}),\n",
```
Comment (JoanFM) [inline basic-basic-crud-functions.ipynb]: this cannot work

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: basic-basic-crud-functions.ipynb (hunk 2 above)
Comment (Kelton8Z) [reply basic-basic-crud-functions.ipynb]: good catch 😅

**Comment by Kelton8Z** [comment]
> also fix the issue with f.block

`with f:
    f.post('/index', DocumentArray([Document(text=t.strip()) for t in open(os.__file__) if t.strip()])) # index all lines of this file
    c = Client(host='localhost', port_expose=12345)  # connect to localhost:12345
    c.post('/search', Document(text='@request(on=something)'), on_done=print_matches)` 
is getting `gRPC error: StatusCode.UNAVAILABLE failed to connect to all addresses` and `BadClient: something wrong when running the eventloop, result can not be retrieved` on notebook, any suggestion for fixes ?

File: basic-basic-crud-functions.ipynb [hunk 3]
```diff
@@ -163,8 +163,8 @@
    "outputs": [],
    "source": [
     "with f:\n",
-    "    f.search(docs[0], top_k=3,\n",
-    "             on_done=lambda x: [print(m.dict()) for m in x.docs[0].matches])"
+    "    f.search(docs[0], parameters={'top_k': 4},,\n",
+    "             on_done=lambda x: [print(m.dict()) for m in x.data.docs[0].matches])"
```
Comment (JoanFM) [inline basic-basic-crud-functions.ipynb]: I think without `data` should work, can u check?

**Review by JoanFM — CHANGES_REQUESTED** [review]

**Comment by bwanglzu** [comment]
not sure, but do we also need to update readme in core?

File: basic-basic-crud-functions.ipynb (hunk 3 above)
Comment (cristianmtr) [reply basic-basic-crud-functions.ipynb]: For me this notebook fails:

```python
           pod0@70478[C]:can not load the executor from _index
           pod0@70478[E]:ExecutorFailToLoad() during <class 'jina.peapods.runtimes.zmq.zed.ZEDRuntime'> initialization
 add "--quiet-error" to suppress the exception details
Traceback (most recent call last):
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/jina/peapods/runtimes/zmq/zed.py", line 91, in _load_executor
    runtime_args=vars(self.args),
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/jina/jaml/__init__.py", line 553, in load_config
    return JAML.load(tag_yml, substitute=False)
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/jina/jaml/__init__.py", line 89, in load
    r = yaml.load(stream, Loader=JinaLoader)
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/yaml/__init__.py", line 114, in load
    return loader.get_single_data()
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/yaml/constructor.py", line 51, in get_single_data
    return self.construct_document(node)
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/yaml/constructor.py", line 55, in construct_document
    data = self.construct_object(node)
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/yaml/constructor.py", line 100, in construct_object
    data = constructor(self, node)
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/yaml/constructor.py", line 429, in construct_undefined
    node.start_mark)
yaml.constructor.ConstructorError: could not determine a constructor for the tag '!_index'
  in "<unicode string>", line 1, column 1:
    !_index
    ^

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/jina/peapods/peas/__init__.py", line 152, in run
    runtime = self._build_runtime()
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/jina/peapods/peas/__init__.py", line 138, in _build_runtime
    timeout_ctrl=self._timeout_ctrl,
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/jina/peapods/runtimes/zmq/zed.py", line 59, in __init__
    self._load_executor()
  File "/home/cristian/envs/notebooks/lib/python3.7/site-packages/jina/peapods/runtimes/zmq/zed.py", line 104, in _load_executor
    raise ExecutorFailToLoad from ex
```

File: basic-feed-data.ipynb [hunk 4]
```diff
@@ -112,17 +112,8 @@
...
-     "collapsed": false
-    },
-    "source": []
-   }
   }
  },
  "nbformat": 4,
  "nbformat_minor": 4
-}
+}
```
Comment (cristianmtr) [inline basic-feed-data.ipynb:120]: this notebook also fails

```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/tmp/ipykernel_71003/2230952711.py in <module>
      2 
      3 with f:
----> 4     f.index(Document.from_ndarray(numpy.random.random([4,2]), on_done=print))  # index ndarray data, document sliced on first dimension

AttributeError: type object 'Document' has no attribute 'from_ndarray'
```

File: decentralized-flow.ipynb [hunk 5]
```diff
@@ -83,4 +83,4 @@
  },
  "nbformat": 4,
  "nbformat_minor": 4
-}
+}
```
Comment (cristianmtr) [inline decentralized-flow.ipynb:85]: this notebook hangs. @deepankarm Is the cloud.jina.ai still available?

**Review by cristianmtr — CHANGES_REQUESTED** [review]

**Comment by Kelton8Z** [comment]
`f.add(uses='docker://jinahub/pretrained-cnn:latest')` from the old hub is working, but I don't think there is a 2.0 hub equivalent ?

File: basic-basic-crud-functions.ipynb (hunk 3 above)
Comment (Kelton8Z) [reply basic-basic-crud-functions.ipynb]: replaced 1.x _index builtin indexer with NumpyPostgresSearcher

File: decentralized-flow.ipynb (hunk 5 above)
Comment (deepankarm) [reply decentralized-flow.ipynb:85]: cloud.jina.ai is not available as the terraform module is broken after jinad 2.0 changes.

File: basic-feed-data.ipynb (hunk 4 above)
Comment (Kelton8Z) [reply basic-feed-data.ipynb:120]: Good catch, fixed

File: decentralized-flow.ipynb (hunk 5 above)
Comment (cristianmtr) [reply decentralized-flow.ipynb:85]: @Kelton8Z then remove this notebook for now. Or add a comment saying it doesn't work at the moment, and point them to the JinaD cookbook to get them started on their own

Comment (Kelton8Z) [reply decentralized-flow.ipynb:85]: marked as WIP

File: basic-basic-crud-functions.ipynb (hunk 3 above)
Comment (Kelton8Z) [reply basic-basic-crud-functions.ipynb]: > I think without `data` should work, can u check?

yes it works

**Comment by bwanglzu** [comment]
- [x] 2.0-readme-38l
- [x] basic-add-logic
- [x]  basic-async-flow
- [x] basic-construct-document
- [x] basic-create-flow
- [ ] basic-crud-functions need to bump searcher version
- [x] basic-feed-data
- [x] basic-fetch-result
- [x] basic-inter-intra-parallel
- [x] basic-visualize-flow
- [ ] basic-decentralize-flow not working dependent on jinaD

**Comment by cristianmtr** [comment]
```
Running decentralized-flow : 0 / 11
Error executing the notebook "decentralized-flow".
See notebook "decentralized-flow_out" for the traceback.
Running basic-inter-intra-parallelism : 1 / 11
Running basic-feed-data : 2 / 11
Running basic-visualize-a-flow : 3 / 11
Running basic-add-logic : 4 / 11
Running basic-crud-functions : 5 / 11
Error executing the notebook "basic-crud-functions".
See notebook "basic-crud-functions_out" for the traceback.
Running basic-construct-document : 6 / 11
Running basic-async-flow : 7 / 11
Error executing the notebook "basic-async-flow".
See notebook "basic-async-flow_out" for the traceback.
Running basic-fetch-result : 8 / 11
Error executing the notebook "basic-fetch-result".
See notebook "basic-fetch-result_out" for the traceback.
Running basic-create-flow : 9 / 11
Running 2.0-readme-38l : 10 / 11
```

FYI you can batch run notebooks with this script https://ogden.eu/run-notebooks

**Review by bwanglzu — APPROVED** [review]
LGTM, create another issue to fix dependencies in other repos, and wrap up this ticket first.

**Review by JoanFM — APPROVED** [review]

**Pull request merged**